File: cpp/src/observatory/operation/RdmaWriteThroughputOperation.cpp
#include "RdmaWriteThroughputOperation.h"

namespace Observatory {

RdmaWriteThroughputOperation::RdmaWriteThroughputOperation(Benchmark *benchmark, Benchmark::Mode mode, uint32_t operationCount, uint32_t operationSize) :
        RdmaThroughputOperation(benchmark, mode, operationCount, operationSize, Benchmark::RdmaMode::WRITE) {}

const char* RdmaWriteThroughputOperation::getClassName() const {
    return "RdmaWriteThroughputOperation";
}

const char* RdmaWriteThroughputOperation::getOutputFilename() const {
    return "RDMA Write Throughput";
}

}

File: java/benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/RdmaWriteThroughputOperation.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.Connection.Mode;
import de.hhu.bsinfo.observatory.benchmark.Connection.RdmaMode;

public class RdmaWriteThroughputOperation extends RdmaThroughputOperation {

    RdmaWriteThroughputOperation(Connection connection, Mode mode, int operationCount, int operationSize) {
        super(connection, mode, operationCount, operationSize, RdmaMode.WRITE);
    }

    @Override
    String getOutputFilename() {
        return "RDMA Write Throughput";
    }
}
